### PR TITLE
SONARJAVA-1475 Assign super field in constructor

### PR DESCRIPTION
**Pull request by benzonico**

### CODE REVIEW

File: java-checks/src/main/java/org/sonar/java/checks/SuperClassFieldInConstructorCheck.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+      }
+
+      if (idTree != null && isFieldFromSuperClass(idTree)) {
+        reportIssue(idTree, "Invoke the \"super\" constructor that sets this field instead.");
+      }
+    }
+
+    private boolean isFieldFromSuperClass(IdentifierTree idTree) {
+      Type idOwnerType = idTree.symbol().owner().type();
+      return owner.type() != idOwnerType && owner.type().isSubtypeOf(idOwnerType);
```
Comment (SonarTech) [inline java-checks/src/main/java/org/sonar/java/checks/SuperClassFieldInConstructorCheck.java]: MAJOR Change this comparison to use the equals method. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1698)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Comment by benzonico** [comment]
This rule is generating too much noise and won't make it in production.

**Pull request closed without merging**